### PR TITLE
Export button

### PR DESCRIPTION
**Pull request by Guillaume28**
addition of the export function

### CODE REVIEW

File: service-monitoring/data.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ function loadPage()
             jQuery("#serviceMonitoringTable").empty().append(htmlData);
             var hostMonitoringTable = jQuery("#serviceMonitoringTable").find("img, style, script, link").load(function () {
                 var h = document.getElementById("serviceMonitoringTable").scrollHeight + 50;
-                parent.iResize(window.name, h);
+                ResizeFrame(window.name, h);
```
Comment (kduret) [inline service-monitoring/data.js]: Why rename method ?

File: service-monitoring/data.js [hunk 2]
```diff
@@ -50,4 +50,12 @@ jQuery(function () {
...
     });
-});
\ No newline at end of file
+});
+
+function ResizeFrame(ifrm, height) {
+    if (height < 150) {
+        height = 150;
+    }
+    console.log(jQuery('[name="' + ifrm + '"]'));
```
Comment (kduret) [inline service-monitoring/data.js]: Remove console.log

File: service-monitoring/src/export.php [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+$query .= " AND h.name NOT LIKE '_Module_%' ";
+$query .= " AND s.enabled = 1 ";
+if (isset($preferences['host_name_search']) && $preferences['host_name_search'] != "") {
+    $tab = split(" ", $preferences['host_name_search']);
+    $op = $tab[0];
+    if (isset($tab[1])) {
+        $search = $tab[1];
+    }
+    if ($op && isset($search) && $search != "") {
+        $query = CentreonUtils::conditionBuilder($query, "h.name ".CentreonUtils::operandToMysqlFormat($op)." '".$dbb->escape($search)."' ");
```
Comment (kduret) [inline service-monitoring/src/export.php]: more than 120 characters in a line

File: service-monitoring/src/export.php [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+    }
+}
+if (isset($preferences['service_description_search']) && $preferences['service_description_search'] != "") {
+    $tab = split(" ", $preferences['service_description_search']);
+    $op = $tab[0];
+    if (isset($tab[1])) {
+        $search = $tab[1];
+    }
+    if ($op && isset($search) && $search != "") {
+        $query = CentreonUtils::conditionBuilder($query, "s.description ".CentreonUtils::operandToMysqlFormat($op)." '".$dbb->escape($search)."' ");
```
Comment (kduret) [inline service-monitoring/src/export.php]: more than 120 characters in a line

File: service-monitoring/src/toolbar.php [hunk 5]
```diff
@@ -1,154 +1 @@
...
                                });
            } else {
                alert("<?php echo _('Please select one or more items'); ?>"); 
                return false;
            }
            $(".toolbar").val(0);
        }
    });
});
</script>
```
Comment (kduret) [inline service-monitoring/src/toolbar.php]: Bad encode

**Comment by lpinsivy** [comment]
feature already present in widget

**Pull request closed without merging**